### PR TITLE
BLS signatures implementation

### PR DESCRIPTION
**Pull request by SonicFrog**
This adds BLS as a signing algorithm for aggregate signatures

### CODE REVIEW

File: src/crypto/bls.rs [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+    {
+        use serde::de::Visitor;
+
+        struct ByteVisitor;
+
+        impl<'de> Visitor<'de> for ByteVisitor {
+            type Value = BlsPrivateKey;
+
+            fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.write_str("byte representation of a bls public key")
```
Comment (tharvik) [inline src/crypto/bls.rs:218]:
```suggestion
                f.write_str("byte representation of a bls private key")
```

**Review by tharvik — APPROVED** [review]

**Comment by tharvik** [comment]
For reference on BLS not being design for multi-signature of a single message:
* IETF: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-3.1
* kyber/bls: https://github.com/dedis/kyber/blob/master/sign/bls/bls.go

**Comment by tharvik** [comment]
Hum, by reread the IETF draft, it seems that it can be made secure using a [proof of possession](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-3.3), you might want to look into it.

**Comment by SonicFrog** [comment]
Thanks for the heads-up I'll look around to see if I can find some proof of possession library

**Pull request merged**